### PR TITLE
Addresses design review feedback

### PR DESCRIPTION
**Pull request by pipwilson**
Addresses feedback listed in https://dsdmoj.atlassian.net/browse/SP2-615

### CODE REVIEW

File: server/views/partials/back-to-top-link.njk [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<a href="#top" class="back-to-top-link">
+<a href="#top" class="back-to-top-link govuk-link govuk-link--no-visited-state">
     <svg role="presentation" focusable="false" class="back-to-top-link__icon" xmlns="http://www.w3.org/2000/svg" width="13" height="17" viewBox="0 0 13 17">
```
Comment (alexw-moj) [inline server/views/partials/back-to-top-link.njk:2]: `presentation` is not a valid svg role. https://www.unimelb.edu.au/accessibility/techniques/accessible-svgs

File: server/routes/editGoal/locale.json [hunk 2]
```diff
@@ -13,7 +14,9 @@
     "dateSelection": {
       "label": "When should {{ subject.givenName }} achieve this?",
       "hint": "Select one option",
-      "customRadioOptionText": "Something else",
+      "customRadioOptionText": "Set another date",
+      "customOptionLabel": "Enter a custom date",
+      "customOptionHint": "For example, 27 3 2007",
```
Comment (alexw-moj) [inline server/routes/editGoal/locale.json:19]: Why `2007`? Do we intend to use these somewhere? They aren't currently used.

File: server/views/pages/plan-summary.njk [hunk 3]
```diff
@@ -42,7 +42,7 @@
         {% elseif data.status == 'updated' %}
             {{ mojBanner({
             type: 'success',
-            text: 'You changed a goal from '+data.popData.givenName+'’s plan',
+            text: 'You changed a goal in '+data.popData.givenName+'’s plan',
```
Comment (alexw-moj) [inline server/views/pages/plan-summary.njk]: Should use the new `possessiveName` instead?

File: server/views/partials/back-to-top-link.njk (hunk 1 above)
Comment (Thomas-Geraghty) [reply server/views/partials/back-to-top-link.njk:2]: This was copied from alpha-gov.

File: server/services/sentence-plan/referentialDataService.ts [hunk 4]
```diff
@@ -7,6 +7,19 @@ export default class ReferentialDataService {
...
+      if (a.name < b.name) {
+        return -1
+      }
+
+      if (a.name > b.name) {
+        return 1
+      }
+
+      return 0
+    })
```
Comment (Thomas-Geraghty) [inline server/services/sentence-plan/referentialDataService.ts]: I'm pretty sure this can just be `this.getAreasOfNeed().sort()`

Comment (izzyL-DJ) [reply server/services/sentence-plan/referentialDataService.ts]: For some reason when we tried just that it didn't work, then we had to put in the custom code

Comment (Thomas-Geraghty) [reply server/services/sentence-plan/referentialDataService.ts]: Might need to be `this.getAreasOfNeed().sort((a, b) => a.name.localeCompare(b.name))`

Comment (Thomas-Geraghty) [reply server/services/sentence-plan/referentialDataService.ts]: resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/107/commits/1305e230b1058239fe77f34f2f08c6e4aa4c9737, thanks @izzyL-DJ `

**Review by Thomas-Geraghty — APPROVED** [review]

**Pull request merged**